File: 백준/Silver/1515. 수 이어 쓰기/수 이어 쓰기.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const N = input[0]

let start = 1
let index = 0

while (index < N.length) {
  const str = start.toString()

  for (let i = 0; i < str.length; i += 1) {
    if (str[i].includes(N[index])) {
      index += 1
    }
  }

  if (index >= N.length) {
    break
  }
  start += 1
}
console.log(start)

File: 백준/Silver/1515. 수 이어 쓰기/수 이어 쓰기.py
import sys
input = sys.stdin.readline


target = list(input().strip())

num = 0
index =0

while index<len(target):
  num+=1
  strNum = str(num)
  for i in strNum:
    if(i == target[index]):
      index+=1
      if(index == len(target)):
          break

print(num)